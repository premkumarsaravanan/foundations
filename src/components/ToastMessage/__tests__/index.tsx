import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ToastMessage, ToastMessageProps, ToastVariant } from '..'

const defaultProps: ToastMessageProps = {
  visible: true,
  message: 'Toast message',
  variant: 'primary',
  onCloseToast: jest.fn()
}

describe('ToastMessage', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<ToastMessage {...defaultProps} />))).toMatchSnapshot()
  })

  it('should dismiss toast when click', () => {
    shallow(<ToastMessage {...defaultProps} />)
      .find('[data-test="toast-wrapper"]')
      .first()
      .simulate('click')

    expect(defaultProps.onCloseToast).toHaveBeenCalledTimes(1)
  })

  it('should dismiss toast after 3 seconds', () => {
    jest.useFakeTimers()
    shallow(<ToastMessage {...defaultProps} />)
    jest.runAllTimers()
    expect(defaultProps.onCloseToast).toHaveBeenCalledTimes(1)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
