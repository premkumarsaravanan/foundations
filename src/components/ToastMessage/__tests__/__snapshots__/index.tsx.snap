// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage should match a snapshot 1`] = `
<div
  className="toast visible"
  data-test="toast-wrapper"
  onClick={[MockFunction]}
>
  <Component
    fullWidth={true}
    type="reset"
    variant="primary"
  >
    Toast message
  </Component>
</div>
`;
